src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,54 @@
+package duke.task;
+
+/**
+ * Represents a task.
+ */
+public abstract class Task {
+    private String name;
+    private boolean isDone;
+
+    protected Task(String name) {
+        this(name, false);
+    }
+
+    protected Task(String name, boolean isDone) {
+        this.name = name;
+        this.isDone = isDone;
+    }
+
+    protected Task markAsDone() {
```

Comment:
Maybe method name should be setDone? As coding standard said setter methods for boolean variables should be of that form.

Reply:
I personally think `setDone` would be a more appropriate name for an actual setter method i.e. 
```java
protected Task setDone(boolean isDone) {
    this.isDone = isDone;
}
```
, which may be implemented in future. Hence I chose the name `markAsDone` so that we don't confuse the methods.

Thanks for the suggestion though!